Revise main documentation page (#8019)
kubernetes/ingress-nginx: docs/index.md

> # NGINX Ingress Controller

Author: Alternatives:
- Leave `# Welcome`
- Switch to `# Overview`

Author: This will revise the heading on https://kubernetes.github.io/ingress-nginx/

Reviewer: I like # Overview

This is the documentation for the NGINX Ingress Controller.

> It is built around the [Kubernetes Ingress resource](http://kubernetes.io/docs/user-guide/ingress/), using a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) to store the controller configuration.

Reviewer 2: ```suggestion
It is built around the [Kubernetes Ingress resource](https://kubernetes.io/docs/concepts/services-networking/ingress/), using a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/) to store the controller configuration.
```

You can learn more about using [Ingress](http://kubernetes.io/docs/user-guide/ingress/) in the official [Kubernetes documentation](https://docs.k8s.io).

## Getting Started

See [Deployment](./deploy/) for a whirlwind tour that will get you started.

# FAQ - Migration to apiVersion `networking.k8s.io/v1`

If you are using Ingress objects in your cluster (running Kubernetes older than v1.22), and you plan to upgrade to Kubernetess v1.22, this section is relevant to you.

- Please read this [official blog on deprecated Ingress API versions](https://kubernetes.io/blog/2021/07/26/update-with-ingress-nginx/) 

- Please read this [official documentation on the IngressClass object](https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-class)

## What is an IngressClass and why is it important for users of Ingress-NGINX controller now ?

IngressClass is a Kubernetes resource. See the description below.

> Its important because until now, a default install of the Ingress-NGINX controller did not require any IngressClass object. From version 1.0.0 of the Ingress-NGINX Controller, a IngressClass object is required.

Reviewer 2: ```suggestion
Its important because until now, a default install of the Ingress-NGINX controller did not require any IngressClass object. From version 1.0.0 of the Ingress-NGINX Controller, an IngressClass object is required.
```

     for this field. For more information, refer to the IngressClass
     documentation.
```

The `.spec.ingressClassName` behavior has precedence over the deprecated `kubernetes.io/ingress.class` annotation.

> ## I have only one instance of the Ingresss-NGINX controller in my cluster. What should I do ?

Reviewer 2: ```suggestion
## I have only one instance of the Ingress-NGINX controller in my cluster. What should I do ?
```

- If you have only one instance of the Ingress-NGINX controller running in your cluster, and you still want to use IngressClass, you should add the annotation `ingressclass.kubernetes.io/is-default-class` in your IngressClass, so that any new Ingress objects will have this one as default IngressClass.

> In this case, you need to make your controller aware of the objects. If you have any Ingress objects that don't yet have either the [`.spec.ingressClassName`](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/ingress-v1/#IngressSpec) field set in their manifest, or the ingress annotation (`kubernetes.io/ingress.class`), then you should start your NGINX ingress controller with the flag `--watch-ingress-without-class=true`.

Reviewer 2: ```suggestion
In this case, you need to make your controller aware of the objects. If you have any Ingress objects that don't yet have either the [`.spec.ingressClassName`](https://kubernetes.io/docs/reference/kubernetes-api/service-resources/ingress-v1/#IngressSpec) field set in their manifest, or the ingress annotation (`kubernetes.io/ingress.class`), then you should start your Ingress-NGINX controller with the flag `--watch-ingress-without-class=true`.
```

## I am seeing this error message in the logs of the Ingress-NGINX controller: "ingress class annotation is not equal to the expected by Ingress Controller". Why ?

- It is highly likely that you will also see the name of the ingress resource in the same error message. This error messsage has been observed on use the deprecated annotation (`kubernetes.io/ingress.class`) in a Ingress resource manifest. It is recommended to use the `.spec.ingressClassName` field of the Ingress resource, to specify the name of the IngressClass of the Ingress you are defining.

## How to easily install multiple instances of the ingress-NGINX controller in the same cluster ?
- Create a new namespace
  ```
  kubectl create namespace ingress-nginx-2
  ```
- Use Helm to install the additional instance of the ingress controller

> - Ensure you have Jelm working (refer to the [Helm documentation](https://helm.sh/docs/))

Reviewer: ```suggestion
- Ensure you have Helm working (refer to the [Helm documentation](https://helm.sh/docs/))
```

- We have to assume that you have the helm repo for the ingress-NGINX controller already added to your Helm config. But, if you have not added the helm repo then you can do this to add the repo to your helm config;
  ```
  helm repo add ingress-nginx https://kubernetes.github.io/ingress-nginx
  ```
- Make sure you have updated the helm repo data;
  ```
  helm repo update
  ```
- Now, install an additional instance of the ingress-NGINX controller like this ;
  ```
  helm install ingress-nginx-2 ingress-nginx/ingress-nginx  \
  --namespace ingress-nginx-2 \
  --set controller.ingressClassResource.name=nginx-two \

>   --set controller.ingressClassResource.controllerValue="example.com/ingress-nginx-2" \

Reviewer: ```suggestion
  --set controller.ingressClassResource.controllerValue="example.com/ingress-nginx-two" \
```

Author: Actually should be `example.com/ingress-nginx2` to match earlier text.
